### ISSUE TITLE
why my infer result from pretrained German-English 4 layer NMT + GNMT model is poor

### ISSUE THREAD

**Issue by xueyanyun** [tensorflow/nmt#290]
I download pretrained model  from http://download.tensorflow.org/models/nmt/ende_gnmt_model_4_layer.zip
code branch tf-1.4:
![image](https://user-images.githubusercontent.com/6800736/38228374-a55ae384-3735-11e8-8840-549838c30242.png)
tensorflow version:1.4.1
![image](https://user-images.githubusercontent.com/6800736/38228344-72791468-3735-11e8-8d72-206322b9a864.png)
my  command for loading the pre-trained GNMT WMT English-German checkpoint for inference.
python -m nmt.nmt \
--src=en --tgt=de \
--ckpt=ende_gnmt_model_4_layer/translate.ckpt \
--hparams_path=nmt/standard_hparams/wmt16_gnmt_4_layer.json \
--out_dir=ende_gnmt \
--vocab_prefix=wmt16/vocab.bpe.32000 \
--inference_input_file=wmt16/newstest2015.tok.bpe.32000.en \
--inference_output_file=ende_gnmt/output_infer \
--inference_ref_file=wmt16/newstest2015.tok.bpe.32000.de

the result is :
2018-04-03 11:11:20.765841: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
2018-04-03 11:11:20.976262: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1030] Found device 0 with properties:
name: Tesla P40 major: 6 minor: 1 memoryClockRate(GHz): 1.531
pciBusID: 0000:83:00.0
totalMemory: 22.38GiB freeMemory: 5.97GiB
2018-04-03 11:11:20.976322: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1120] Creating TensorFlow device (/device:GPU:0) -> (device: 0, name: Tesla P40, pci bus id: 0000:83:00.0, compute capability: 6.1)
  loaded infer model parameters from ende_gnmt_model_4_layer/translate.ckpt, time 1.22s
# Start decoding
  decoding to output ende_gnmt/output_infer.
^@^@2018-04-03 11:14:28.590557: W tensorflow/core/framework/op_kernel.cc:1192] Out of range: End of sequence
	 [[Node: IteratorGetNext = IteratorGetNext[output_shapes=[[?,?], [?]], output_types=[DT_INT32, DT_INT32], _device="/job:localhost/replica:0/task:0/device:CPU:0"](Iterator)]]
2018-04-03 11:14:28.591408: W tensorflow/core/framework/op_kernel.cc:1192] Out of range: End of sequence
	 [[Node: IteratorGetNext = IteratorGetNext[output_shapes=[[?,?], [?]], output_types=[DT_INT32, DT_INT32], _device="/job:localhost/replica:0/task:0/device:CPU:0"](Iterator)]]
  done, num sentences 2169, num translations per input 1, time 186s, Tue Apr  3 11:14:28 2018.
  **bleu: 11.3**

why the bleu is so low? thanks

**Comment by mohammedayub44** [comment]
I'm getting more worst results my test and dev blue metric shows as 0.0. I replicated the whole setup on windows machine. Using the pre-trained 8 layer model. Not sure if there is something missing in the setup doc. I followed this link. https://www.tensorflow.org/tutorials/seq2seq
